dev/shell.nix

```diff
@@ -1,16 +1,15 @@
+{ pkgs, ... }:
```

Comment:
A general observation: as a reader I don't know what the interface is until I see in what context that file is being consumed.

This is a common issue already between nixpkgs callPackage and NixOS modules. But this is making things worse.

I don't have a really good answer except maybe that each file might have to be annotated on the top with `# flake-parts perSystem module` or some qualifier like that.

Reply:
I don't see how this is a problem? The top level flake.nix provides context, don't see any benefit in adding comments to files?

Reply:
It's sub-optimal because the reader doesn't have all the context when opening the file. In order to find out, they have to now search where the file is being used. It also makes the life of editors harder to provide auto-complete.

As we become more experienced with Nix, we learn to recognize those patterns and learn to infer that a file starting with `{ stdenv, ... }:`, for example, is probably loaded using `callPackage`.

But it's just a general observation and a particularity of Nix, not something we necessarily need to act upon.

Reply:
I'm really struggling to understand your point of view.

I'd say https://github.com/nix-community/infra/pull/518 and `build0*/default.nix` parts of https://github.com/nix-community/infra/pull/584 also removed context?

Honestly my take away from you comments is flattening files and splitting up the flake is bad?